### ISSUE TITLE
Control the transitive dependency resolution on a granular level similar to ^ and ~

### ISSUE THREAD

**Issue by alexzaytsev-newsroomly** [NuGet/Home#6770]
Remove the content above here and fill out details below.

## Details about Problem

NuGet product used (NuGet.exe | VS UI | Package Manager Console | dotnet.exe):
dotnet.exe
NuGet version (x.x.x.xxx):
4.6.0.4964

dotnet.exe --version (if appropriate):
2.1.100

VS version (if appropriate):
15.6.0

OS version (i.e. win10 v1607 (14393.321)):
win10 v1703 (15063.966)

Worked before? If so, with which NuGet version:
No

## Detailed repro steps so we can see the same problem

consider the following dependency tree:

Project 1
- Library
- - Library 2 (1.0.0)
- Library 2 (1.0.0)

Now if my `Library 1` updates `Library 2` dependency even to a minor version (e.g. `1.0.1`) and I update `Library 1 `in my project, i get the annoying `NU1605 Detected package downgrade error` (that used to be a warning).

Is there anything like npm's `^version` operator that would let me reference `1.0.*` in my `Project 1 `so i don't have to rebuild `Library 1` each time i want to update `Library 2`?

**Comment by nkolev92** [comment]
NU1605 is still a warning. 
Somewhere in your project, the `<TreatWarningsAsErrors>true</TreatWarningsAsErrors>` is set. 
That could potentially be set by the SDK, etc. 
You can do /pp or use the bl to figure out where it's set and decide if you want to change it or not. 

From Project 1's perspective, the closest thing to `^` in npm, is a floating version, but that comes with some caveats. 
Currently there's no dependency locking for floating versions which means you're always going to get the latest version. We're on working on it in the following [task](https://github.com/NuGet/Home/issues/5602).

When you pack it however, it'll pack the currently used version and that'll be considered a minimum. 
NuGet's [resolution ](https://docs.microsoft.com/en-us/nuget/consume-packages/dependency-resolution#dependency-resolution-rules) strategy for transitive dependencies is lowest applicable. 

Basically, there's [no way](https://github.com/NuGet/Home/issues/5553#issuecomment-337728425) to float the dependencies of Library1 when library1 is added to the Project1. 

Related [issue](https://github.com/NuGet/Home/issues/5553) that talks about the resolution strategy of transitive dependencies. 
That is consumer controlled however. 

//cc @rrelyea @jainaashish @anangaur for visibility

**Comment by nkolev92** [comment]
Since https://github.com/NuGet/Home/issues/6848 got more engagement, I'm closing this one out as a duplicate of that one.
